### ISSUE TITLE
multiple tabs in an editor

### ISSUE THREAD

**Issue by amrit-poudel** [spyder-ide/spyder#3014]
This comment is not about an "issue" in Spyder, but more about its feature.

I was just wondering whether it was possible to arrange multiple tabs in a single editor in a vertical way (instead of a horizontal way that is currently available) so that it is easy to see which files are open. I can't seem to find this feature in Spyder. Matlab editor and other editors like sublime etc., have this feature and is very convenient when we want to edit multiple files. 

Thanks.

**Comment by ccordoba12** [comment]
Could you post an screenshot of what you want?

**Comment by amrit-poudel** [comment]
![multipletabs](https://cloud.githubusercontent.com/assets/12847183/13382198/1a159c4e-de34-11e5-9a9f-852172bdf100.png)

Here is a screenshot of Matlab editor. On the left you will notice multiple tabs with respective file names. The tab position can be chosen anywhere (top, bottom, left or right).

**Comment by goanpeca** [comment]
we will need to make a custom tab widget to hhave more freedom on the way we locate the tabs. can be done...

**Comment by amrit-poudel** [comment]
I can give it a try on my own, but am not sure where to start from.

Also, I like to be able to change the position/location of the file in the tab, just like in matlab.

**Comment by goanpeca** [comment]
@amrit-poudel could you post some screenshots of the different things that you would like to see (matlab screenshots)

**Comment by goanpeca** [comment]
Also what is the behavior when you have more files open than what fits on available vertical screen?

**Comment by ccordoba12** [comment]
@goanpeca, i don't think it's useful to pursue this now that we have the new file switcher.

**Comment by goanpeca** [comment]
Probably, just wanted to have more context ;-)

**Comment by amrit-poudel** [comment]
Thanks for the update and added new features. However, I  think file switcher is not as efficient as having multiple tabs. This is  my (user) perspective.

As for  @goanpeca 's question, if we have more files open,  we can put a scroll bar that can allow us to scroll up and down.

It appears to me that the most recent version of spyder (v 3) does not seem to offer a choice to user where to put the tabs. The tabs are placed horizontally on the top, which is not always convenient.

**Comment by DanCodigaMWRA** [comment]
I saw that (as long as I am understanding correctly) the ability to drag/drop reorder tabs (when there are multiple files/tabs open in the editor) has made it on to the roadmap for Spyder 3.1 and this is great news!! I'm a former Matlab user of many many years and have now transitioned to Spyder over the past ~year or so. I am using Spyder 3.0.2 now and this feature (drag/drop reordering of editor tabs) is IMHO one of the primary weaknesses of Spyder-- it impedes my productivity pretty much on a daily basis. It would be VERY helpful to have this implemented finally.

**Comment by ccordoba12** [comment]
@DanCodigaMWRA, yep, reordering tabs (but not having tabs organized vertically, which this is issue is about) will come in 3.1 :-)

Thanks for your patience.

**Comment by HeiniK** [comment]
Vertical tabs, or multiline horizontal tabs would also be of great practical use to me. If i have more than about 6-8 files openend in the editor, I always have to scroll back and forth through the displayed tabs.

File switching by Ctrl+page up/page down can be quite a pain in the eye, when browsing through 15-20 files.

**Comment by jnsebgosselin** [comment]
@HeiniK Have you tried using the fileswitcher (<kbd>Ctrl</kbd>+<kbd>P</kbd>)? If yes, why is it not solving this issue and how could it be improved?

The fileswitcher works like an improved version of a vertical tabbar with search capabilities and with the added benefit of not cluttering the interface, especially when working split editors. It can also be used to browse through the outline tree of a file with <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>P</kbd> or by adding `@` in the search bar.

![fileswitcher_demo](https://user-images.githubusercontent.com/10170372/35645456-da2466ac-0699-11e8-95eb-ad55070625ce.gif)

**Comment by amrit-poudel** [comment]
For users who are used to Matlab, file switch is not very efficient since you have to constantly do back and forth/ scroll up and down, as the above user has also pointed out. It may be efficient if you have, say 100 files open, but that is very rare.  Having a vertical tab, just like in Matlab, is very handy, at least for those doing scientific computing where it is common to work with a fewer number of files.

**Comment by jnsebgosselin** [comment]
@amrit-poudel thanks for the inputs. There is also the dropdown menu on the left side of the tabbar that can be used to quickly navigate between files.

![image](https://user-images.githubusercontent.com/10170372/35660249-f3ad9e9c-06d8-11e8-9d2e-5a92b637a239.png)

But I understand it is not the same thing as having a permanent and fully functional vertical tabbar like in Matlab. How does it looks in Matlab when the editor is split? Let's say something like in the image below. Isn't there a lot of wasted space due to the vertical tabbar?

![image](https://user-images.githubusercontent.com/10170372/35660489-42dc04d0-06da-11e8-8ff3-bfb037f3f86d.png)

**Comment by m-beau** [comment]
Up, any update on this front? This would be an awesome feature!
